SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -200,31 +203,73 @@ private Task ProcessPendingErrorsAsync()
 
         private Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            return HandleUserConfirmationAsync();
+            bool alwaysSend = ApplicationSettings.GetValue(PrefKeyAlwaysSend, false);
+
+            return Task.Run(async () =>
+            {
+                if (_unprocessedManagedErrorLogs.Count() > 0)
+                {
+                    // Check for always send: this bypasses user confirmation callback.
+                    if (alwaysSend)
+                    {
+                        AppCenterLog.Debug(LogTag, "The flag for user confirmation is set to AlwaysSend, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                        return;
+                    }
+
+                    if (ShouldAwaitUserConfirmation?.Invoke() ?? false)
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned true, wait sending logs.");
+                    }
+                    else
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned false, will send logs.");
```

Comment:
Can this be simplified by combining this block with the first block for always send? That way the method only invokes `HandleUserConfirmationAsync(UserConfirmation.Send);` in a single place

Reply:
It could, but this implementation exactly mirrors the Android implementation. Unless you feel strongly, I'll keep it this way.